images/router/haproxy/reload-haproxy

```diff
@@ -6,24 +6,40 @@ config_file=/var/lib/haproxy/conf/haproxy.config
 pid_file=/var/lib/haproxy/run/haproxy.pid
 old_pid=""
 haproxy_conf_dir=/var/lib/haproxy/conf
-readonly max_retries=30
+readonly max_wait_time=10
+readonly timeout_opts="-m 1 --connect-timeout 1"
+readonly numeric_re='^[0-9]+$'
 
-function waitForHAProxyToStartListening() {
+function haproxyHealthCheck() {
+  local wait_time=${MAX_RELOAD_WAIT_TIME:-"$max_wait_time"}
   local port=${STATS_PORT:-"1936"}
   local retries=0
+  local start_ts=$(date +"%s")
 
-  echo " - Checking if HAProxy is listening on port $port ..."
-  while ! lsof -nPp $(< $pid_file) | grep ":${port} (LISTEN)" &> /dev/null ; do
-    sleep 0.2
-    retries=$((retries + 1))
-    if [ $retries -ge $max_retries ]; then
-      echo " - Exceeded $max_retries retries waiting for HAProxy to start listening on port $port"
-      return
+  if ! [[ $wait_time =~ $numeric_re ]]; then
+    echo " - Invalid max reload wait time, using default $max_wait_time ..."
+    wait_time=$max_wait_time
+  fi
+
+  local end_ts=$((start_ts + wait_time))
+
+  echo " - Checking HAProxy /healthz on port $port ..."
+  while true; do
+    local httpcode=$(curl $timeout_opts -s -o /dev/null -I -w "%{http_code}" http://localhost:${port}/healthz)
```

Comment:
The default is to bind `:1936` - which is all interfaces. 
Whilst it is possible for someone to listen to only a set of interfaces, it would require a custom haproxy router image [change the template config].